### PR TITLE
Tezos: add voting support

### PR DESCRIPTION
**Pull request by adonagy**
Added two new protobuf messages to tezos for voting - proposal and ballot respectively.

### CODE REVIEW

File: protob/messages-tezos.proto [hunk 1]
```diff
@@ -120,6 +122,24 @@ message TezosSignTx {
...
+        optional bytes proposals = 4;
+    }
+    /**
+     * Structure representing information for ballot
+     */
+    message TezosBallotOp {
+        optional bytes source = 1;
+        optional uint64 period = 2;
+        optional bytes proposal = 3;
+        optional bytes ballot = 4;
```
Comment (matejcik) [inline protob/messages-tezos.proto]: reading the core PR, this field is a single byte encoding "yay/nay/pass"
that should be an enum then

File: protob/messages-tezos.proto [hunk 2]
```diff
@@ -120,6 +122,24 @@ message TezosSignTx {
         optional uint64 storage_limit = 5;
         optional bytes delegate = 6;
     }
+    /**
+     * Structure representing information for proposal
+     */
+    message TezosProposalOp {
+        optional bytes source = 1;
```
[the remark this answers is not in the document]
Comment (adonagy) [reply protob/messages-tezos.proto]: You are rigth. Will change that.

File: protob/messages-tezos.proto [hunk 3]
```diff
@@ -120,6 +122,24 @@ message TezosSignTx {
...
+        optional uint64 period = 2;
+        optional uint64 bytes_in_next_field = 3;
+        optional bytes proposals = 4;
+    }
+    /**
+     * Structure representing information for ballot
+     */
+    message TezosBallotOp {
+        optional bytes source = 1;
```
[the remark this answers is not in the document]
Comment (adonagy) [reply protob/messages-tezos.proto]: You are rigth. Will change that.

File: protob/messages-tezos.proto [hunk 4]
```diff
@@ -120,6 +122,24 @@ message TezosSignTx {
         optional uint64 storage_limit = 5;
         optional bytes delegate = 6;
     }
+    /**
+     * Structure representing information for proposal
+     */
+    message TezosProposalOp {
+        optional bytes source = 1;
+        optional uint64 period = 2;
+        optional uint64 bytes_in_next_field = 3;
```
[the remark this answers is not in the document]
Comment (adonagy) [reply protob/messages-tezos.proto]: It comes with the message from the tezos node (http://tezos.gitlab.io/mainnet/api/p2p.html#proposals-tag-5). But yeah, using `len(proposals)` makes more sense.

File: protob/messages-tezos.proto (hunk 2 above)
[the remark this answers is not in the document]
Comment (adonagy) [reply protob/messages-tezos.proto]: Correction: the `source` field in ballot and proposal != `source` field as a `TezosContractID`

In messages like transaction, revelation, etc. the source is indeed the contract ID, but in ballot and proposal operations the source filed contains only the `public_key_hash` as opposed to other source fields which contain an additional `tag` field. 

See (http://tezos.gitlab.io/mainnet/api/p2p.html#transaction-tag-8) vs. (http://tezos.gitlab.io/mainnet/api/p2p.html#proposals-tag-5)

[the remark this answers is not in the document]
Comment (matejcik) [reply protob/messages-tezos.proto]: ok, please add a comment that this is intentional :)

**Comment by prusnak** [comment]
Superseded by https://github.com/trezor/trezor-firmware/pull/41

**Pull request closed without merging**